### ISSUE TITLE
Format problem with coloured terminal output on windows

### ISSUE THREAD

**Issue by PierreEVEN** [xmake-io/xmake#1924]
The picture bellow is quite explicit i think.... Sometime (50% chances to happen) the colour format is broken. (CMD + PowerShell)

> Maybe a hidden character that broke the colour format ?

### Describe the bug

![image](https://user-images.githubusercontent.com/24438631/146639204-68a33b41-68ed-4b72-950b-91445fe8fc51.png)

### Expected behavior

![image](https://user-images.githubusercontent.com/24438631/146639283-3c37fa2e-68db-43ad-8a4c-a484c5107357.png)

*(not the same text, but you got the idea. Sometimes it works.. sometime not)*

I have the problem with two different computers. It is easier to reproduce **when installing third party dependencies** for the first time.

### Related Environment

- xmake version: 2.6.2 (dev). Also happen on latest release
- os: Windows 10 (CMD and Powershell)

### Note

Not realy a priority of course, but it would be nice to have a clean output 🙃
(maybe the problem is really dumb)

**Comment by PierreEVEN** [comment]
## Update 

This happen everytime i do a clean install (remove the .xmake cache)

At first beginning it works fine, but sudently the output just blew up.

![image](https://user-images.githubusercontent.com/24438631/146639650-2c3dbf43-8002-4f4e-9eab-b306a1f8891e.png)

**Comment by waruqi** [comment]
The `git submodule update` command will destroy it. At present, I can only alleviate this problem, but I cannot avoid it completely, https://github.com/xmake-io/xmake/issues/1501#issuecomment-877273250
https://github.com/xmake-io/xmake/commit/0d01a974ac4ffcf85f6349dc899c4b83635a951e

**Comment by PierreEVEN** [comment]
Woops okay thanks !
Sorry for the duplicated post 😅

**Comment by SirLynix** [comment]
Maybe this could be fixed by calling git in a external process (with its own stdout), or maybe even in its own terminal if required

**Comment by waruqi** [comment]
git will call SetConsoleMode to modify terminal mode, when downloading in parallel, xmake's coroutine will switch to other tasks to print colors, which will be broken at this time.

https://github.com/git/git/blob/9d530dc0024503ab4218fe6c4395b8a0aa245478/compat/terminal.c#L110

**Comment by waruqi** [comment]
> Maybe this could be fixed by calling git in a external process (with its own stdout), or maybe even in its own terminal if required

Call an external process or terminal, which may be ok, but it seems a bit heavy. I can try to save and restore terminal output mode

**Comment by waruqi** [comment]
I try to fix it. @see https://github.com/xmake-io/xmake/pull/1926

You can update to `term_mode` branch and try it again.

```console
$ xmake update -s github:xmake-io/xmake#term_mode
```

**Comment by PierreEVEN** [comment]
I tried rinstalling three times, no error. Great job, thanks ! 🙂
